apps/vmq_server/src/vmq_queue.erl

```diff
@@ -1060,15 +1084,27 @@ insert(MsgOrRef, #state{id = SId, deliver_mode = balance, sessions = Sessions} =
     State#state{sessions = maps:update(RandomPid, UpdatedSession, Sessions)}.
 
 session_insert(SId, #session{status = active, queue = Q} = Session, MsgOrRef) ->
-    {send(SId, Session#session{queue = queue_insert(false, MsgOrRef, Q, SId)}), MsgOrRef};
+    {
+        send(SId, Session#session{
+            queue_to_session_batch_size = 100, queue = queue_insert(false, MsgOrRef, Q, SId)
```

Comment:
hard coded batch size, wondering what to base this on. The session process can send a lot of messages to the consumer, depending on max_inflight/receive_max_client (in case QoS >1).

Reply:
Thanks. That is actualy a left-over from testing/development. The standard batch size of "100" was in there before. I will remove that line and it will revert to whatever queue_to_session_batch_size was.  

BatchSize = maps:get(queue_to_session_batch_size, Opts, 100),